routes/github/commits.ts

```diff
@@ -14,12 +14,17 @@ export default async function route(req: Request, res: Response) {
     const msg = "optional query parameter 'to' must be a single numeric string";
     return res.status(400).send(msg);
   }
+  if (typeof path !== "undefined" && typeof path !== "string") {
+    res.set("Content-Type", "text/plain");
+    const msg = "optional query parameter 'path' must be a string";
+    return res.status(400).send(msg);
+  }
 
```

Comment:
Need some more validation on the path, as path is included in cache key (corresponding to a filesystem path (see [`DiskCache`](https://github.com/jnurthen/respec-web-services/blob/e9e9c9219fbe7c5925150756a3d1f474d44e4c59/utils/disk-cache.ts))).
I think we should normalize the path (resolve all `..`) and ensure it starts with a `./`, so that we don't end up reading files out of scope of cache area.

Rest looks good.

Reply:
Done in https://github.com/speced/respec-web-services/pull/440